### ISSUE TITLE
Performance issue when using a static response variable

### ISSUE THREAD

**Issue by mgoldyan** [jersey/jersey#3644]
In Jersey 2.x , I used to keep a reference to a **static variable**: 
`final static EMPTY_RESPONSE = Response.noContent().build();`

When I load test (using JMeter), I simply return the static variable for an incoming HTTP (GET) request and the performance is very low compared to generating the response per request (using the ResponseBuilder).

Here's the GET API call: 
```
@GET
@Path(DEBUG_PATH + "/empty")
public Response getEmptyResponse() { return EMPTY_RESPONSE; }
```

The CPU goes to 100% in both cases, but the performance **drops from 30K RPS to 150 only** when using the static variable.

BTW, I don't see this issue in Jersey 1.x.

**Comment by pavelbucek** [comment]
I did rerun some tests locally with similar results. And then, I understood what is happening..

.. build response has a state. And note that response can be closed (and it actually IS closed once the processing is done), which makes sense, since Response can contain an entity and if that entity is an OutputStream, it is generally not repeatable.

I'm not exactly sure what is happening in your case (.noContent()), but if you'd try to set an entity, you'll see that each consequent response will fail with `java.lang.IllegalStateException: The output stream has already been closed.`

So, if you want to cache something, please cache last instance of ResponseBuilder, just before `.build()` is invoked. You'll get better results.

I'm not closing this issue, since it is still worth investigating why is the difference so notable. On the other side, I don't take it as high priority, since I consider caching a built Response object a bad practice for reasons stated above.

**Comment by mgoldyan** [comment]
Pavel, 
First of all, thank you for the quick response.

It's interesting why this doesn't happen in Jersey 1.19 (low performance happens in Jersey 2.25).
I couldn't find anywhere in the documentation of Jersey 2 a warning about the ResponseBuilder generating a stateful instance or about it isn't repeatable.

It seems like there's some kind of a R/W lock in Jersey 2 since the performance keeps dropping as the load test progresses.

Another thing I saw in Jersey 2 might be related is when I used a static final variable for status code 200 with a simple String as its entity:
```
final static String VERSION = "1.0.3";
final static VERSION_RESPONSE = Response.ok().entity(VERSION).build();
```
Returning this (GET) response seems to work for the first few seconds. Afterwards, it "forgets" the entity and returns an empty OK response.

Do you think the two issues are related...?

Thanks in advance.

**Comment by mgoldyan** [comment]
@pavelbucek , anything new on this matter?
10x

**Comment by pavelbucek** [comment]
@mgoldyan no. As stated previously, don't expect this to be fixed soon, since it is a bad practice from Jersey perspective to do what you are doing.

Response object has a state and is not supposed to be reused for multiple responses (take in mind the input stream scenario).
